vault/seal.go

```diff
@@ -191,6 +204,11 @@ func (d *defaultSeal) SetBarrierConfig(ctx context.Context, config *SealConfig)
 
 	config.Type = d.BarrierType()
 
+	if d.core.isRaftUnseal() {
```

Comment:
Can you put in a comment here explaining why you're only storing the config if it's Raft?

Reply:
@vishalnayak Should we be only storing this on raft or all the time?

Reply:
I think this was done to ensure that we don't try to persist the barrier config to storage while the Join is in progress. At the time, Vault would still be sealed and the storage call would fail. So, we just store the barrier config in memory and return.

But I wonder if this path will be hit after unseal to actually persist the barrier config.

Reply:
I just added a comment to this effect.